packages/hoprd/crates/hoprd-api/src/lib.rs

```diff
@@ -977,12 +995,29 @@ mod messages {
 
         if let Some(path) = &args.path {
             if path.len() > 3 {
-                return Ok(Response::builder(422).body(ApiErrorStatus::UnknownFailure("The path components must contain at most 3 elements".into())).build())
+                return Ok(Response::builder(422)
+                    .body(ApiErrorStatus::UnknownFailure(
+                        "The path components must contain at most 3 elements".into(),
+                    ))
+                    .build());
             }
         }
 
-        match hopr.send_message(Box::from(args.body.as_ref()), args.peer_id, args.path, args.hops, Some(args.tag)).await {
-            Ok(challenge) => Ok(Response::builder(202).body(json!(SendMessageRes{challenge})).build()),
+        let timestamp = std::time::SystemTime::now()
```

Comment:
Are we interested in a timestamp when the `send_message` was initiated or when it was sent to the first hop (not acknowledged) ?

Reply:
Either one will work.
The reason for this is just not to rely on the clients clock but rather get a fix point from the node that can be used to query responses.